src/System.Diagnostics.PerformanceCounter/tests/PerformanceCounterCategoryTests.cs

```diff
@@ -0,0 +1,368 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Specialized;
+using Xunit;
+
+namespace System.Diagnostics.Tests
+{
+    [SkipOnTargetFramework(TargetFrameworkMonikers.Uap)] // In appcontainer, cannot write to perf counters
+    public static class PerformanceCounterCategoryTests
+    {
+        [ConditionalFact(typeof(AdminHelpers), nameof(AdminHelpers.IsProcessElevated))]
+        public static void PerformanceCounterCategory_CreatePerformanceCounterCategory_DefaultConstructor()
+        {
+            PerformanceCounterCategory pcc = new PerformanceCounterCategory();
+            Assert.Equal(".", pcc.MachineName);
+        }
+
+        [ConditionalFact(typeof(AdminHelpers), nameof(AdminHelpers.IsProcessElevated))]
+        public static void PerformanceCounterCategory_CreatePerformanceCounterCategory_NullTests()
+        {
+            Assert.Throws<ArgumentNullException>(() => new PerformanceCounterCategory(null, "."));
+            Assert.Throws<ArgumentException>(() => new PerformanceCounterCategory(string.Empty, "."));
+            Assert.Throws<ArgumentException>(() => new PerformanceCounterCategory("category", string.Empty));
+        }
+
+        [ConditionalFact(typeof(AdminHelpers), nameof(AdminHelpers.IsProcessElevated))]
+        public static void PerformanceCounterCategory_SetCategoryName_Valid()
+        {
+            PerformanceCounterCategory pcc = new PerformanceCounterCategory();
+            pcc.CategoryName = "Processor";
+            Assert.Equal("Processor", pcc.CategoryName);
+        }
+
+        [ConditionalFact(typeof(AdminHelpers), nameof(AdminHelpers.IsProcessElevated))]
```

Comment:
It's not a big deal, but many of these tests don't need this attribute.

Reply:
BTW you could replace it with 
`[ConditionalFact(typeof(Helpers), nameof(Helpers.CanWriteToPerfCounters))]`
or similar, akin to what event logs did with `[ConditionalFact(typeof(Helpers), nameof(Helpers.SupportsEventLogs))]`

Nice thing about this is it makes a single place you can tune when the tests run later.

Reply:
And it appears you need to -- your tests are failing on Windows Nano. So once you centralize as I usggest, you'll need to add `&& !PlatformDetection.IsWindowsNanoServer` to the check.